airbyte-cdk/python/airbyte_cdk/destinations/vector_db_based/document_processor.py

```diff
@@ -139,22 +143,27 @@ def _extract_relevant_fields(self, record: AirbyteRecordMessage, fields: Optiona
 
     def _extract_metadata(self, record: AirbyteRecordMessage) -> Dict[str, Any]:
         metadata = self._extract_relevant_fields(record, self.metadata_fields)
+        metadata[METADATA_STREAM_FIELD] = create_stream_identifier(record)
+        primary_key = self._extract_primary_key(record)
+        if primary_key:
+            metadata[METADATA_RECORD_ID_FIELD] = primary_key
+        return metadata
+
+    def _extract_primary_key(self, record: AirbyteRecordMessage) -> Optional[str]:
         stream_identifier = create_stream_identifier(record)
         current_stream: ConfiguredAirbyteStream = self.streams[stream_identifier]
-        metadata[METADATA_STREAM_FIELD] = stream_identifier
         # if the sync mode is deduping, use the primary key to upsert existing records instead of appending new ones
-        if current_stream.primary_key and current_stream.destination_sync_mode == DestinationSyncMode.append_dedup:
-            metadata[METADATA_RECORD_ID_FIELD] = f"{stream_identifier}_{self._extract_primary_key(record, current_stream)}"
-        return metadata
+        if not current_stream.primary_key or not current_stream.destination_sync_mode == DestinationSyncMode.append_dedup:
```

Comment:
Nit... Generally != or "is not" would be preferable for the "not equals" comparison.